src/OpenLoco/src/Ui/Windows/Construction/ConstructionTab.cpp

```diff
@@ -2622,22 +2676,71 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
         constructionLoop(constructPos, maxRetries, constructHeight);
     }
 
+    static void onToolUpMultiple(Window& self, const WidgetIndex_t widgetIndex)
+    {
+        mapInvalidateMapSelectionTiles();
+        removeConstructionGhosts();
+
+        auto posA = World::TilePos2{ std::min(_toolPosInitial.x, _toolPosDrag.x), std::min(_toolPosInitial.y, _toolPosDrag.y) };
+        auto posB = World::TilePos2{ std::max(_toolPosInitial.x, _toolPosDrag.x), std::max(_toolPosInitial.y, _toolPosDrag.y) };
+
+        auto rotation = _cState->constructionRotation;
+        auto piece = _cState->lastSelectedTrackPiece;
+
+        for (auto tilePos : TilePosRangeView(posA, posB))
```

Comment:
As pointed out by @LeeSpork, this approach may lead to unintuitive results if finances are tight, considering the starting corner is always the same.

Perhaps we should indeed start at `_toolPosInitial` and work towards `_toolPosDrag`, using a regular x/y for loop. This would entail calculating the number of steps in either dimension, and whether the step size is +1 or -1, but nothing too difficult.

Any thoughts on this?

Reply:
I've gone ahead and implemented it like this.